handlers/index.js

```diff
@@ -0,0 +1,199 @@
+const Session = require("../session/session");
+const SessionStorage = require("../session/session_storage");
+const { v4: uuidv4 } = require("uuid");
+const logger = require("../logger");
+const {
+    FdkSessionNotFoundError,
+    FdkInvalidOAuthError,
+} = require("../error_code");
+
+const fpInstall = async (company_id, application_id, ext) => {
+    let companyId = parseInt(company_id);
+    let platformConfig = ext.getPlatformConfig(companyId);
+    let session;
+
+    session = new Session(Session.generateSessionId(true));
+
+    let sessionExpires = new Date(Date.now() + 900000); // 15 min
+
+    if (session.isNew) {
+        session.company_id = companyId;
+        session.scope = ext.scopes;
+        session.expires = sessionExpires;
+        session.access_mode = "online"; // Always generate online mode token for extension launch
+        session.extension_id = ext.api_key;
+    } else {
+        if (session.expires) {
+            session.expires = new Date(session.expires);
+        }
+    }
+
+    let redirectUrl;
+
+    session.state = uuidv4();
+
+    // pass application id if received
+    let authCallback = ext.getAuthCallback();
+    if (application_id) {
+        authCallback += "?application_id=" + application_id;
+    }
+
+    // start authorization flow
+    redirectUrl = platformConfig.oauthClient.startAuthorization({
+        scope: session.scope,
+        redirectUri: authCallback,
+        state: session.state,
+        access_mode: "online", // Always generate online mode token for extension launch
+    });
+    await SessionStorage.saveSession(session);
+    logger.debug(`Redirecting after install callback to url: ${redirectUrl}`);
+    return {
+        redirectUrl: redirectUrl,
+        fdkSession: session,
+    };
+};
+
+const fpAuth = async (reqObj, state, code, ext, sessionId) => {
```

Comment:
Add named function here as it will be helpful while reading stack trace.  @jinalvirani

Reply:
Resolved